common/taskana-common/src/main/java/pro/taskana/common/internal/util/ObjectAttributeChangeDetector.java

```diff
@@ -26,51 +28,22 @@ private ObjectAttributeChangeDetector() {}
    * @throws SystemException when any parameter is null
    */
   public static <T> String determineChangesInAttributes(T oldObject, T newObject) {
-    List<Field> fields = new ArrayList<>();
-
-    if (Objects.isNull(oldObject) || Objects.isNull(newObject)) {
+    if (oldObject == null || newObject == null) {
       throw new SystemException(
           "Null was provided as a parameter. Please provide two objects of the same type");
     }
```

Comment:
should  we additionally check that oldObject and newObject are really the same class? The bound type parameter `T` only verifies that they are compatible to each other. oldObject could be a subclass of newObject. Currently we would generate more changes than they really might be..

Reply:
That's right, even though I would expect the dev who uses this to provide two objects of the same class. Also it's debatable if we would actually want to compare subclasses also. If that's the case, then generating the changes wouldn't be an error in my opinion.

Reply:
alright. Then we keep it like this ;)